Create 2014-11-11-responsive-images-in-practice.md

Justineo opened the pull request:
init

Justineo committed:
Create 2014-11-11-responsive-images-in-practice.md

init

otakustay commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,341 @@
+
+设计师们面临这样一个苏菲的选择：让页面在有些情况下变模糊、还是在所有情况下变慢。大多数人倾向于后者，给所有人发送能适配最大屏幕、最高分辨率的图片。浪费啊。
+
+但是！经过三年的辩论，一些新的用来解决响应式图片这个问题的标记已经出现了：
```
建议搞成：

> 我们有了一些新的标记来解决响应式图片这个问题

Justineo committed:
Update _config.yml

make things right

Justineo committed:
Update 2014-11-11-responsive-images-in-practice.md

added footnotes

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+
+[Eric Portis](http://alistapart.com/author/eportis)
+
+> 魔鬼因一切我们享受的事物而惩罚我们。
```
> 我们必须为享乐付出代价。

所有评注表示提供一种翻译选择，下同。

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+* `srcset`
+* `sizes`
+* `picture`
+* 还有我们的老朋友 `source`（借用自 `audio` 和 `video`）
```
> 源于，来自？

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+* `picture`
+* 还有我们的老朋友 `source`（借用自 `audio` 和 `video`）
+
+这些新元素和新属性让我们可以编写多个可替换的源图片，然后给每个客户端提供最合适的那一个。它们已经进入了官方规范，而且其第一个完整实现已经在九月由 Chrome 38 给出了。再借助优雅的回退策略以及一个用来跨越障碍的 [polyfill](http://scottjehl.github.io/picturefill/)，我们可以并且应当*立刻*实现响应式图片。那么，走起！
```
> 而且在九月 Chrome 38 第一个完整实现了。

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+
+这些新元素和新属性让我们可以编写多个可替换的源图片，然后给每个客户端提供最合适的那一个。它们已经进入了官方规范，而且其第一个完整实现已经在九月由 Chrome 38 给出了。再借助优雅的回退策略以及一个用来跨越障碍的 [polyfill](http://scottjehl.github.io/picturefill/)，我们可以并且应当*立刻*实现响应式图片。那么，走起！
+
+我们先找个已有的网页，来把它的图片做成响应式的。我们要这么来三遍，每次轮番用上新标记的其中一个：
```
> 我们先找个网页来把它的图片做成响应式的。
> 我们先找个网页来实施。
> 
> 我们按以下三步，依次用上一个新标记：

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+
+我们先找个已有的网页，来把它的图片做成响应式的。我们要这么来三遍，每次轮番用上新标记的其中一个：
+
+我们将用 `srcset` 和 `sizes` 来保证我们的图片高效地缩放。
```
貌似丢了序号？

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+
+我们将通过 `picture` 和 `source media` 对我们的图片进行艺术指导。
+
+我们将用 `picture` 和 `source type` 给图片提供可替换格式。
```
三步中的 `我们将` 和 `我们的` 都可以省略吧？

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+
+—[本杰明•富兰克林](http://thinkexist.com/quotation/i_guess_i_don-t_so_much_mind_being_old-as_i_mind/200747.html)（还是[彼得•加布里埃尔](http://www.tejvan.co.uk/funny/funny_quotes_old_age/view/index.html)来着？）
+
+我们把我们的主角设定成[一个关于疯狂拼布的小网页](http://alistapart.com/d/407/demo/status-quo.html)[^2]。这是一个简单且响应式的页面。其中除了主要内容——巨大的图片（拼布的图片！）——以外没太多别的东西了。我们想同时展示每块拼布的整体设计以及尽可能复杂的细节。那么，我们给每块拼布呈现两幅图片：
```
> 我们的主角是..

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+[^2]: 所谓的疯狂拼布，是指在不规则拼接的蚕丝布或天鹅绒布上，用刺绣、珠子、缎带、蕾丝等进行装饰。
+
+1. 整块拼布，和段落同宽；
+2. 一副充满 100% 视口宽度的细节图。
```
> 一幅

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+
+![Chrome 的开发工具显示元素宽度为 620px。](http://alistapart.com/d/407/eric-portis-images-oct/box-model.png)
+
+（我更倾向于第二种方法。）
```
> 我更倾向作弊方法

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+
+无论用哪种方法，我们最终得到了整块拼布的 `img` 最大显示宽度为 620px。我们将以两倍大小渲染我们的源图片来适应双倍屏：1240 像素宽。
+
+但是我们怎么处理我们的细节图片呢？它们要放大充满整个视口的宽度，而视口尺寸没有固定的上限。那我们就选[一个大大的、感觉标标准准的尺寸](http://en.wikipedia.org/wiki/1080p)来渲染，比如说最大 1920 像素宽，就它了。
```
> 但是怎么处理我们的细节图片呢？

去掉第一个 `我们`

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+
+但是我们怎么处理我们的细节图片呢？它们要放大充满整个视口的宽度，而视口尺寸没有固定的上限。那我们就选[一个大大的、感觉标标准准的尺寸](http://en.wikipedia.org/wiki/1080p)来渲染，比如说最大 1920 像素宽，就它了。
+
+当我们按这些尺寸渲染我们的图片时，[我们使用既有方式实现的这个网页](http://alistapart.com/d/407/demo/status-quo.html)重达 3.5MB。其中除了 5.7kB 以外全是图片。我们不用想也知道其中很多的图片字节在发送到小屏、低分屏设备时将成为无形的开销。但有多少呢？让我们开工吧。
```
> 当我们按这些尺寸渲染图片时

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+当我们按这些尺寸渲染我们的图片时，[我们使用既有方式实现的这个网页](http://alistapart.com/d/407/demo/status-quo.html)重达 3.5MB。其中除了 5.7kB 以外全是图片。我们不用想也知道其中很多的图片字节在发送到小屏、低分屏设备时将成为无形的开销。但有多少呢？让我们开工吧。
+
+
+## 第一遍：用 `srcset` 和 `sizes` 缩放图片
```
> 第一步
> 第一轮
> 第一巡

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+
+怎么选择呢？这是我最爱 `srcset` 的一点了：不知道！我们*没法*知道。选择逻辑是故意没有指定的。
+
+这第一个解决响应式图片问题的方案试图给作者*更多*的控制权。这样*我们*就掌管了全局，构建覆盖所有情况的媒体查询——为每一种屏幕大小及分辨率的组合制定应急计划，分别为每一个给出定制的源图片。
```
媒体查询要不要翻译呢....

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+
+`srcset` 把我们解放了出来。合适粒度的控制在我们需要时（等下会说）仍然可用，但是大多数情况下我们最好还是把关键问题移交给浏览器来做决定。浏览器对一个用户的屏幕、视口、网络连接和偏好设置都非常了解。通过出让控制权——描述图片而非为无数个目的地指定特定的源图片——我们使得浏览器能充分利用它已知的那些信息。我们则从少得多的代码中获得了更好的功能（还是对未来友好的！）。
+
+然而，这里有一个坑：选择适当的源图片需要知道图片的布局尺寸。但是我们无法让浏览器把选择过程延迟到页面的 HTML、CSS 和 JavaScript 全部加载且解析完毕以后。所以我们得用另一个新属性来给浏览器提供一个预估的图片显示宽度：`sizes`。
```
> 必须知道图片的布局尺寸才能选择合适的源图片。但是我们无法让浏览器等页面的 HTML、CSS 和 JavaScript 全部加载且解析完毕以后再作选择。

chriswong commented on _drafts/2014-11-11-responsive-images-in-practice.md:
```diff
@@ -0,0 +1,351 @@
+
+幸运的是，`sizes` 可以让我们做两件事：
+
+1. 让我们用一个逗号分隔的列表提供多个长度；
```
省略 `让我们`

Justineo committed:
Update 2014-11-11-responsive-images-in-practice.md

Justineo committed:
Update 2014-11-11-responsive-images-in-practice.md

revised according to comments from @myst
adopt block title syntax of `kramdown`

Justineo committed:
update wording

Justineo committed:
publish

Justineo committed:
move images to local directory

Justineo committed:
try syntax highlight

Justineo committed:
try site.url config

Justineo committed:
another try...

Justineo committed:
try not explicitly config url

Justineo committed:
oops, revert

Justineo committed:
add meta

Justineo committed:
fix typo

Justineo committed:
mod tpl

Justineo committed:
try redcarpet

Justineo committed:
format a little bit

Justineo committed:
use github config instead of hardcoded one

Justineo committed:
try coderay

Justineo committed:
try coderay

Justineo committed:
switch back to kramdown

Justineo committed:
switch back to kramdown

Justineo committed:
fix url

Justineo committed:
try coderay

Justineo committed:
try coderay

Justineo committed:
revert config

Justineo committed:
change img ref

Justineo committed:
update styles

Justineo committed:
fix tab

Justineo committed:
fix styles

Justineo committed:
add hljs

Justineo committed:
add hljs

Justineo committed:
add hljs

Justineo committed:
add hljs

Justineo committed:
remove unused config

Justineo committed:
adjust link style

Justineo committed:
refine wording

Justineo committed:
refine wording

Justineo committed:
add disqus

Justineo committed:
try toc and optimize heading ids

Justineo committed:
try toc

Justineo committed:
remove toc

Justineo committed:
id generation

Justineo committed:
id generation

Justineo committed:
id generation

Justineo committed:
id generation

Justineo committed:
try duoshuo

Justineo committed:
try Han.css

Justineo committed:
try Han.css

Justineo committed:
Han.css maybe not suitable :(

Justineo committed:
Han.css maybe not suitable :(

Justineo committed:
@supports test

Justineo committed:
fix font style

Justineo committed:
add line-height

Justineo committed:
better fallback

Justineo committed:
try uyan
